tests/installation/ipxe_install.pm

```diff
@@ -225,9 +243,13 @@ sub run {
         if (check_screen(\@tags, $ssh_vnc_wait_time)) {
             save_screenshot;
             sleep 2;
+            return if is_usb_boot;
             prepare_disks if (!is_upgrade && !get_var('KEEP_DISKS'));
         }
-        save_screenshot;
+        else {
+            save_screenshot;
+            die "Do not catch needle with tag $ssh_vnc_tag!" if is_usb_boot;
```

Comment:
```Did not catch ...```

Reply:
This is to let test die at the file it firstly fails. Ipxe_install takes care of the part of such needle assertion. So I hope to keep it here. Then about having assert_screen again in usb_install is that sometimes when we need to debug the file with some shortened test flow of commenting off ipxe_boot, it will help.

Reply:
I do not think you need else branch still.

Reply:
I'd like to keep above explained intention here, if you do not mind. Just let it go, a minor thing :)

Reply:
I think you can drop the else branch.